src/busctl/busctl.c

```diff
@@ -1691,9 +1692,25 @@ static int message_append_cmdline(sd_bus_message *m, const char *signature, char
                         break;
                 }
 
-                case SD_BUS_TYPE_UNIX_FD:
-                        return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
-                                               "UNIX file descriptor not supported as type.");
+                case SD_BUS_TYPE_UNIX_FD: {
+                        int fd;
+
+                        r = safe_atoi(v, &fd);
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to parse '%s' as a file descriptor: %m", v);
+
+                        if (!*fdset) {
+                                r = fdset_new_fill(/* filter_cloexec= */ 0, fdset);
+                                if (r < 0)
+                                        return log_error_errno(r, "Failed to create fd set: %m");
+                        }
+
+                        if (!fdset_contains(*fdset, fd))
+                                return log_error_errno(SYNTHETIC_ERRNO(EBADF), "Failed to find file descriptor '%s' in available file descriptors: %m", v);
```

Comment:
do not output `%m` here: you are generating an error here, not displaying info about an error you received after all.